src/run_solver.jl

```diff
@@ -13,6 +14,7 @@ Apply a solver to a set of problems.
   CUTEst problems).
 
 #### Keyword arguments
+* `use_threads::Bool`: whether to use threads (default: `true`);
```

Comment:
Why do we need this keyword? Threads should be enabled/disabled with the environment variable.

Reply:
I think the `use_threads` keyword is valuable to keep, even with the environment variable, for the following reasons:

Testing and Debugging: This flag allows us to explicitly override the environment variable during testing or debugging. It’s particularly useful to verify that the application behaves correctly in both threaded and non-threaded modes without re-writing the functionality in the unit tests.

Fine-Grained User Control: Some users may want to override the global threading setting for specific scenarios:

They might want to disable threading for this application while leaving it enabled for other programs to avoid resource contention.

I think keeping this flag ensures flexibility and adaptability, making the tool more useful for both development and usage scenarios.

Reply:
A user who doesn’t set `JULIA_NUM_THREADS` is not expected anything to run multi-threaded, yet `use_threads` will be `true`. I think that only introduces confusion.